lib/src/main/scala/spinal/lib/eda/fusesoc/FuseSocGeneratorBuilder.scala

```diff
@@ -0,0 +1,68 @@
+package spinal.lib.eda.fusesoc
+
+import spinal.core._
+import com.fasterxml.jackson.core.{JsonFactory, JsonParser}
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.{YAMLFactory, YAMLGenerator}
+import com.fasterxml.jackson.module.scala._
+
+abstract class FuseSocGeneratorBuilder[P, C<:Component](componentName: String, defaultPram: P) {
+  private val yamlMapper = new ObjectMapper(new YAMLFactory().disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER))
+  yamlMapper.registerModule(DefaultScalaModule)
+  private val jsonMapper = new ObjectMapper(new JsonFactory().configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true).configure(JsonParser.Feature.INCLUDE_SOURCE_IN_LOCATION, true))
+  jsonMapper.registerModule(DefaultScalaModule)
+
+
+  private case class P2SParam(spinal_parameter: P,
+                              output: Object = Map("files"->List(Map(componentName+".v"->new Object{val file_type: String = "verilogSource"}))),
+                              target_directory: String = "./generate",
+                              entry_function: String = "",
+                              copy_core: Boolean=false,
+                              spinal_project_path: String = ".")
+
+
+  private val CoreFile = new Object {
+    val name: String = s"::$componentName:0.0.0"
```

Comment:
Is it possible to fill the required version and info below through parameters of class construction method?

Reply:
ok